### PR TITLE
basic opentelemetry integration

### PR DESCRIPTION
**Pull request by ericsullivan**
Thanks for your Pull Request 🎉 

**Please keep these instructions in mind so we can review it more efficiently:**

- Add the references of all the related issues/PRs in the description
- Whether it's a new feature or a bug fix, make sure they're covered by new test cases
- If this PR contains any refactoring work, please give it its own commit(s)
- Finally, please add an entry to the corresponding changelog

**Other Notes**
- We squash all commits before merging
- We generally review new PRs within a week
- If you have any question, you can ask for feedback in our [discord community](https://discord.gg/Ww9hbqr) first

## Description
Describe your changes:

### CODE REVIEW

**Comment by izzergh** [comment]
This is part of an issue in Enrollment:
* https://github.com/annkissam/fms_enrollment_birch/pull/2985
* https://www.flowdock.com/app/annkissam/general-rails-dev/threads/pQxZZ1aPMG2BMI6Nma-wVMuUDK3

Depending on platform, bundler swaps the order of `sentry-ruby` and `sentry-rails` in the Gemfile.lock. If the order disagrees with what bundler determines is correct on the client's server, bundler won't install.

Is there anything you can do with the gemspecs to ensure the order does not change by platform? I realize this is a really open-ended request, but I (a 2019 mac user) am willing to pair with you (a Linux user) on it.

File: sentry-rails/lib/sentry/rails/railtie.rb [hunk 1]
```diff
@@ -7,7 +7,12 @@ class Railtie < ::Rails::Railtie
     # middlewares can't be injected after initialize
     initializer "sentry.use_rack_middleware" do |app|
       # placed after all the file-sending middlewares so we can avoid unnecessary transactions
-      app.config.middleware.insert_after ActionDispatch::ShowExceptions, Sentry::Rails::CaptureExceptions
+      if defined?(OpenTelemetry::Instrumentation::Rack::Middlewares::TracerMiddleware)
+        app.config.middleware.insert_after OpenTelemetry::Instrumentation::Rack::Middlewares::TracerMiddleware, Sentry::Rails::CaptureExceptions
+      else
+        app.config.middleware.insert_after ActionDispatch::ShowExceptions, Sentry::Rails::CaptureExceptions
```
Comment (GregoryDuguies) [inline sentry-rails/lib/sentry/rails/railtie.rb:13]: FYI @ericsullivan had to resolve a merge conflict here

Comment (ericsullivan) [reply sentry-rails/lib/sentry/rails/railtie.rb:13]: ok

Comment (ericsullivan) [reply sentry-rails/lib/sentry/rails/railtie.rb:13]: I never tried to push this code upstream. Did you want to try getting a PR into sentry? I didn't know how they'd like it, as they sorta of have their own thing, but I feel eventually they'll want to support this

**Comment by FlorinPopaCodes** [comment]
I believe there might be an issue with this code and it sets the span to 0000000000000000

**Comment by ericsullivan** [comment]
@FlorinPopaCodes It used to to. Can you get me enough info to reproduce and I'll take a look

**Comment by FlorinPopaCodes** [comment]
When looking at traces for example in https://sentry.bfortress.com/organizations/annkissam/performance/etimesheet_plus:0af25fc9c4314a8ebb474a5e8a098919/?project=31&query=transaction.duration%3A%3C15m+event.type%3Atransaction+http.method%3AGET&statsPeriod=24h&transaction=ShiftsController%23show&unselectedSeries=p100%28%29 

parent_span_id are set to 0000000000000000

Screenshot from 2021-12-14 16-17-26

I was able to reproduce this locally as well in et+ when figuring out traces scoping.

**Comment by FlorinPopaCodes** [comment]
@oneamtu could you close this so I can get my `https://github.com/pulls/mentioned` page to zero? 🙏

**Pull request closed without merging**